src/Base/UnitsSchemaFemMilliMeterNewton.cpp

```diff
@@ -23,37 +23,38 @@
 
 
 #include "PreCompiled.h"
-#ifdef __GNUC__
-#include <unistd.h>
+#ifndef _PreComp_
+#include <algorithm>
+#include <array>
+#include <utility>
 #endif
 
-#include <QString>
-
 #include "UnitsSchemaFemMilliMeterNewton.h"
 
-
 using namespace Base;
 
-
-QString UnitsSchemaFemMilliMeterNewton::schemaTranslate(const Quantity& quant,
-                                                        double& factor,
-                                                        QString& unitString)
+std::string UnitsSchemaFemMilliMeterNewton::schemaTranslate(const Quantity& quant,
+                                                            double& factor,
+                                                            std::string& unitString)
 {
-    Unit unit = quant.getUnit();
-    if (unit == Unit::Length) {
-        // all length units in millimeters
-        unitString = QString::fromLatin1("mm");
-        factor = 1.0;
-    }
-    else if (unit == Unit::Mass) {
-        // all mass units in t
-        unitString = QString::fromUtf8("t");
-        factor = 1e3;
+    static std::array<std::pair<Unit, std::pair<std::string, double>>, 2> unitSpecs {{
+        {Unit::Length, {"mm", 1.0}},
+        {Unit::Mass, {"t", 1e3}},
+    }};
+
+    const auto unit = quant.getUnit();
+    const auto spec = std::find_if(unitSpecs.begin(), unitSpecs.end(), [&](const auto& pair) {
+        return pair.first == unit;
+    });
+
+    if (spec != std::end(unitSpecs)) {
+        unitString = spec->second.first;
+        factor = spec->second.second;
     }
     else {
-        // default action for all cases without special treatment:
         unitString = quant.getUnit().getString();
         factor = 1.0;
     }
+
     return toLocale(quant, factor, unitString);
 }
```

Comment:
This looks like `UnitsSchemaCentimeters::schemaTranslate`. Maybe it could be reworked into some helper static method taking the specs and returning `factor` and `unitString`?

Reply:
Definitely. You'll probably find more places like this. I didn't convert everything and plan to do this as a separate PR. Also note @bofdahof's work which seems very good to me I do not want to add more work to rebase it once this PR is merged.

Reply:
Can you point me to the PR or commit that you are referring to? 

If there is already in progress change (or simply will to address it) that will make this request obsolete it is absolutely fine to leave this as it is improvement. In such case feel free to mark it as resolved.

Reply:
It is not a PR yet, just [a branch](https://github.com/bofdahof/FreeCAD/tree/units). You can find more in conversation history of this PR.

Reply:
Thanks! I must have missed it while scanning over the discussion.

Reply:
Perhaps that _helper static method_ could live in the base class. Or we could find the way of general parametrization and just past the right data when instantiating class.